README.md

```diff
@@ -129,51 +129,89 @@ the `encryption_*` and `ssl_*` options
+`.irodsA` per the convention of iRODS's iCommands):
+   - `irods.client_init.write_native_irodsA_file`
+   - `irods.client_init.write_pam_irodsA_file`
+
+These functions can roughly be described as duplicating the function of `iinit`,
+provided that a valid `irods_environment.json` has already been created.
+
+Each of the above functions can take a cleartext password and write an appropriately encoded
+version of it into an authentication in the appropriate location.  That location is
+`~/.irods/.irodsA` unless the environment variable IRODS_AUTHENTICATION_FILE has been set
+in the command shell to dictate an alternative file path.
+
+As an example, here we write a native `.irodsA` file using the first of the two functions.  We
+provide the one required argument, a password string entered - in this case - interactively at the
+terminal.
+
+```bash
+bash$ echo '{ "irods_user_name":"rods",
+              ... # other parameters as needed
+            }'> ~/.irods/irods_environment.json
+bash$ python -c "import irods.client_init, getpass
+irods.client_init.write_native_irodsA_file(getpass.getpass('Enter iRODS password -> '))"
 ```
-irods.client_init.write_native_credentials_to_secrets_file
-irods.client_init.write_pam_credentials_to_secrets_file
+
+By default, when an `.irodsA` file already exists, it will be overwritten. If however the
+`overwrite` parameter is set to `False`, an exception of type `irods.client_init.irodsA_already_exists`
+is raised to warn of any older `.irodsA` file that might otherwise have been overwritten.
+
+Equivalently to the above, we can issue the following command.
+
+```bash
+bash$ prc_write_irodsA.py native <<<"${MY_CURRENT_IRODS_PASSWORD}"
 ```
 
-Each takes a cleartext password and writes an appropriately processed version of it
-into an .irodsA (secrets) file in the login environment.
+The redirect may of course be left off, in which case the user is prompted for the iRODS password
+and echo of the keyboard input will be suppressed, in the style of `iinit`.  Regardless
+which technique is used, no password will be visible on the terminal during or after input.
+
+For the `pam_password` scheme, typically SSL/TLS must first be enabled to avoid sending data related
+to the password - or even sending the raw password itself - over a network connection in the clear.
 
-Examples:
-For the `native` authentication scheme, we can use the currently set iRODS password to create the .irodsA file directly:
+Thus for `pam_password` authentication to work well, we should first ensure, when setting up the
+client environment, to include within `irods_environment.json` the appropriate SSL/TLS connection
+parameters.  In a pinch, `iinit` can be used to verify this prerequisite is fulfilled,
+as its invocation would then create a valid .irodsA from merely prompting the user for their PAM password.
+
+Once again, this can also be done either using the free function directly:
```

Comment:
Perhaps omit the word "either" here? I realize the sentence continues below, but I had to read it a few times to get it. Maybe it's just me. :)

Reply:
done